src/ObjectList.php

```diff
@@ -362,6 +363,60 @@ private function permute(&$out, $list, $perms = [], $max = null)
     }
 
     //
+    // Serialization and deserialization
+    //
+
+    /**
+     * Turn this object into an array using toArray method on each element if they have it.
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        $out = [];
+        foreach ($this->getArrayCopy() as $key => $value) {
+            if (is_object($value) && method_exists($value, 'toArray')) {
```

Comment:
I think we can optimize it a little and not check on each element if `method_exists`.  Maybe we can check it only once? But maybe we shouldn't think about micro optimization.